### ISSUE TITLE
Have a configuration for charms to always update charm libraries when packing

### ISSUE THREAD

**Issue by facundobatista** [canonical/charmcraft#988]
Add a configuration key in `charmcrat.yaml`, in charmcraft.yaml, to always update the charm libraries before packing.

So, the user could add this line in the config file:

```
charm-libraries-auto-update: true
```

(alternate key names: `charm-libraries-auto-update-when-packing`, `charm-libraries-pack-update`)

and when packing the net result will be that *before* actually packing the `fetch-lib` will be run (applying to all libraries).

The key belongs to the root level, not in the charm part, as this is for Charmcraft to change behaviour before packing and not part of the plugin's behaviour.

If the "fetch lib" command fails, the packing should be interrupted.

**Comment by sed-i** [comment]
This may break CI/charms:
1. A charm has the key set to true
2. As part of merge CI, everything is ok
3. Between merge and release a lib was updated: a charm is released with a new lib version without any testing

How about:
```bash
$ charmcraft pack
WARNING: Outdated library: charm.lib.whatever
```
or
```
$ charmcraft pack
Aborted: Outdated library: charm.lib.whatever

$ charmcraft pack --ignore-outdated-libs
Ok
```

**Comment by facundobatista** [comment]
The "verification during packing and show a message" is already proposed in #213.

The case you described about running tests and then releasing with other version is real. If you have that, do not set the config key :) But yes, it may be problematic...

We can have a flag in `pack`, something like `--update-libraries`.

But if I have to type that for every pack...
```
charmcraft pack --update-libraries
```
*I* would just do everything manually...
```
charmcraft fetch-lib
charmcraft pack
```

What do you think?

**Comment by sed-i** [comment]
Agreed, needing to manually pass `--update-libs` is the same as remembering to `fetch-lib`.

**Comment by PietroPasotti** [comment]
I agree with @sed-i that there's a problem here. I don't think that this should be charmcraft's concern. I agree that we need to somehow automate this (or make it easier), but I don't think charmcraft is the right place to do so.

Alternatives I see:
- in CI: every time a new PR is opened, or on a schedule (daily?): check if any of the libs you use has newer versions. If so, open a PR and an issue.

**Comment by sergiusens** [comment]
I would make this behave more like renovate, either by adding it to renovate itself (https://github.com/renovatebot/renovate/tree/main/lib/modules/datasource) or copying the spirit of it
